electric_emission_cost/costs.py

```diff
@@ -366,16 +383,36 @@ def get_charge_df(
         if any(k in key for k in ["electric_customer", "gas_customer"])
     }
 
+    if scale_fixed_charges or scale_demand_charges:
+        # calculate the scale factor
+        month = start_dt.month
+        year = end_dt.year
+        mins_in_month = (
+            (dt.date(year, month + 1, 1) - dt.date(year, month, 1)).days * 24 * 60
+        )
+        bins_in_month = mins_in_month / res_binsize_minutes
+        scale_factor = ntsteps / bins_in_month
+    else:
+        scale_factor = 1.0
+
     if keep_fixed_charges:
         # replace the fixed charge in charge_dict with its time-averaged value
         for key, value in fixed_charge_dict.items():
-            charge_dict[key] = np.ones(ntsteps) * value / ntsteps
-
+            arr = np.zeros(ntsteps)
+            arr[0] = value[0] * scale_factor
```

Comment:
There's a slight bug here in the edge case where `scale_demand_charges=True` but `scale_fixed_charges=False` and `keep_fixed_charges=True`, the `scale_factor` will be set to a value other than 1 for demand charges but applied to customer charges. Also don't you want to distribute the charges throughout? So I think it should be:

```python
if scale_fixed_charges:
    arr = np.ones(ntsteps) * value[0] * scale_factor
else:
    arr = np.zeros(ntsteps)
    arr[0] = value[0]

charge_dict[key] = arr
```